src/components/DataBar/DataBar.tsx

```diff
@@ -0,0 +1,208 @@
+   * 3) Manage tabIndex to allow focus before setting focus.
+   * 4) Set focus on the next or previous element respective to the arrow key interaction.
+   */
+  /* 1 */
+  const [focusableElementIndex, setFocusableElementIndex] = React.useState(0);
+  const segmentsRef = React.useRef([]); /* 2 */
+  const handleOnKeyDown = (
+    e: React.KeyboardEvent<HTMLElement>,
+    index: number,
+  ) => {
+    if (e.key === 'ArrowLeft' && index > 0) {
+      setFocusableElementIndex(focusableElementIndex - 1); /* 3 */
+      segmentsRef.current[index - 1].focus(); /* 4 */
+    }
+    if (e.key === 'ArrowRight' && index < segmentComponents.length - 1) {
+      setFocusableElementIndex(focusableElementIndex + 1); /* 3 */
+      segmentsRef.current[index + 1].focus(); /* 4 */
+    }
+  };
+
+  /**
+   * Adds a segment component to the segmentComponents array
+   * 1) Keeps a running accumulator to prevent adding any more segments if the accumulated value has already met max.
+   * 2) Ensures a minimumum width of 5% for the segment.
+   * 3) Rounds the right side of the segment if it completes the data bar.
+   * 4) If both the DataBar container and segments specify variants, prioritizes the segment variant.
+   */
+  for (
+    let index = 0, accumulator = 0 /* 1 */;
+    index < segments.length && accumulator < max;
+    index++
+  ) {
+    const segment = segments[index];
+    accumulator += segment.value;
+
+    const percentage = Math.max(5, (segment.value / max) * 100); /* 2 */
+
+    const isRoundRight = accumulator >= max; /* 3 */
+    segmentComponents.push(
+      <DataBarSegment
+        aria-label={segment.text}
+        isRoundRight={isRoundRight}
+        key={`segment-${index}`}
+        onKeyDown={(e) => handleOnKeyDown(e, index)}
+        ref={(el) => (segmentsRef.current[index] = el)}
+        tabIndex={focusableElementIndex === index ? 0 : -1}
+        text={segment.text}
+        variant={segment.variant || variant} /* 4 */
+        width={`${percentage}%`}
+      />,
+    );
+  }
+
+  /**
+   * Adds a miniscule amount of segment as a visual indicator that the component is a data/progress bar and not a pill.
+   */
+  if (!segmentComponents.length) {
+    segmentComponents.push(
+      <DataBarSegment
+        isHoverable={false}
+        key="segment-empty"
+        variant={variant}
+        width={'var(--eds-size-1)'}
+      />,
+    );
+  }
+
+  const componentClassName = clsx(styles['data-bar'], className);
+
+  /**
+   * Leaves space if values do not total to max to visibly show space remaining.
+   */
+  const segmentSpaceClassName = clsx(
+    styles['data-bar__segment-space'],
+    !isFull && styles[`data-bar__segment-space--incomplete`],
+  );
+
+  const id = useUID();
+  const captionId = useUID();
+  const caption = totalSegmentValue + '/' + max;
+  return (
+    <div>
+      <div className={styles['data-bar__label-wrapper']}>
+        <label className={styles['data-bar__label']} htmlFor={id}>
```

Comment:
Didn't use the component `<Label>` since it already had styling and the html element is used simply